src/java_bytecode/java_object_factory.cpp

```diff
@@ -194,10 +500,29 @@ symbolt &new_tmp_symbol(symbol_tablet &symbol_table)
     new_symbol.name="tmp_object_factory$"+std::to_string(++temporary_counter);
     new_symbol.base_name=new_symbol.name;
     new_symbol.mode=ID_java;
-  } while(symbol_table.move(new_symbol, symbol_ptr));
+  }
+  while(symbol_table.move(new_symbol, symbol_ptr));
 
   return *symbol_ptr;
 }
+
+/*******************************************************************\
+
+Function: get_nondet_bool
+
+  Inputs: Desired type (C_bool or plain bool)
+
+ Outputs: nondet expr of that type
+
+ Purpose:
+
+\*******************************************************************/
+
+exprt get_nondet_bool(const typet& type)
+{
+  // We force this to 0 and 1 and won't consider
+  // other values.
+  return typecast_exprt(side_effect_expr_nondett(bool_typet()), type);
```

Comment:
If/when #35 is merged, we should consider re-using its nondet-initialiser infrastructure for this bit of code.